### ISSUE TITLE
Command failed: "npm" -v

### ISSUE THREAD

**Issue by taliesinpenbardd** [TiagoSilvaPereira/vemto-issues#26]
I still can't "run" the project in Vemto because Vemto can't see NPM.

MacOS 10.15.7
Vemto 0.5.5
```
$ node -v
v15.0.1
$ which node
/usr/local/bin/node
$ npm -v
7.0.3
$ which npm
/usr/local/bin/npm
$ /usr/local/bin/npm -v
7.0.3
```

On the settings page, I've tried to leave the NPM field free, or to fill it with `/usr/local/bin` but to no avail. Could there be some sort of caching that prevents the change to be effective?

Is there anything I can tell you about it that could help you to reproduce that?
Thanks!

**Comment by TiagoSilvaPereira** [comment]
Hi @taliesinpenbardd today, after I finished 0.5.5, I had a similar issue... It was gone after rebooting the computer... could you please try rebooting to see if it could fix the problem?

It is hard to know what is happening in that case but I'll research more to understand.

**Comment by taliesinpenbardd** [comment]
I will try to reboot the computer, but in the meantime, I've tried to run NPM from iTerm, and I have an error, that may be the source of the problem:

```
$ npm run watch

> watch
> npm run development -- --watch


> development
> cross-env NODE_ENV=development node_modules/webpack/bin/webpack.js --progress --config=node_modules/laravel-mix/setup/webpack.config.js "--watch"

node:events:304
      throw er; // Unhandled 'error' event
      ^

Error: spawn node_modules/webpack/bin/webpack.js ENOENT
    at Process.ChildProcess._handle.onexit (node:internal/child_process:269:19)
    at onErrorNT (node:internal/child_process:465:16)
    at processTicksAndRejections (node:internal/process/task_queues:80:21)
Emitted 'error' event on ChildProcess instance at:
    at Process.ChildProcess._handle.onexit (node:internal/child_process:275:12)
    at onErrorNT (node:internal/child_process:465:16)
    at processTicksAndRejections (node:internal/process/task_queues:80:21) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'spawn node_modules/webpack/bin/webpack.js',
  path: 'node_modules/webpack/bin/webpack.js',
  spawnargs: [
    '--progress',
    '--config=node_modules/laravel-mix/setup/webpack.config.js',
    '--watch'
  ]
}
npm ERR! code 1
npm ERR! path /Users/user/Sites/donations
npm ERR! command failed
npm ERR! command sh -c cross-env NODE_ENV=development node_modules/webpack/bin/webpack.js --progress --config=node_modules/laravel-mix/setup/webpack.config.js "--watch"

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/user/.npm/_logs/2020-11-02T23_21_22_398Z-debug.log
npm ERR! code 1
npm ERR! path /Users/user/Sites/donations
npm ERR! command failed
npm ERR! command sh -c npm run development -- --watch

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/user/.npm/_logs/2020-11-02T23_21_22_608Z-debug.log
```

Would that help you?

**Comment by TiagoSilvaPereira** [comment]
Hmm... the error above seems like a problem with the node_modules inside the folder you are running the command (https://github.com/JeffreyWay/laravel-mix/issues/1072)... probably doesn't have a relation with the Vemto problem. Is it possible to try reinstalling node and npm and try with Vemto again?

**Comment by taliesinpenbardd** [comment]
Side note: you should trim and escape the settings fields. I tried to add a space after `/usr/local/bin ` and the result is an expected error : `Command failed: "/usr/local/bin\ /npm" -v`

**Comment by TiagoSilvaPereira** [comment]
Thanks, I'll add a trim to the next version... Is it running now or the same issue yet?

**Comment by taliesinpenbardd** [comment]
Same issue still, but that may be because of a dependency conflict on my computer. I'll try and figure it out later.

**Comment by TiagoSilvaPereira** [comment]
Hi @taliesinpenbardd sorry for the late, I just sent **Vemto 0.5.7**. I discovered that on macOS, electron overwrites $PATH environment variable, so it causes this type of errors. I hope it works now... Have a nice day, cheers

**Comment by taliesinpenbardd** [comment]
Well, with this version, Vemto recognizes NPM, that's an improvement! Next error is:
```
Error: Command failed: /usr/local/bin/npm install && /usr/local/bin/npm run dev CLI for webpack must be installed. webpack-cli (https://github.com/webpack/webpack-cli) We will use "npm" to install the CLI via "npm install -D". Do you want to install 'webpack-cli' (yes/no): npm ERR! code 1 npm ERR! path /Users/arthos/Sites/donations npm ERR! command failed npm ERR! command sh -c cross-env NODE_ENV=development node_modules/webpack/bin/webpack.js --progress --config=node_modules/laravel-mix/setup/webpack.config.js npm ERR! A complete log of this run can be found in: npm ERR! /Users/arthos/.npm/_logs/2020-11-04T08_50_44_245Z-debug.log npm ERR! code 1 npm ERR! path /Users/arthos/Sites/donations npm ERR! command failed npm ERR! command sh -c npm run development npm ERR! A complete log of this run can be found in: npm ERR! /Users/arthos/.npm/_logs/2020-11-04T08_50_44_427Z-debug.log

Command failed: /usr/local/bin/npm install && /usr/local/bin/npm run dev CLI for webpack must be installed. webpack-cli (https://github.com/webpack/webpack-cli) We will use "npm" to install the CLI via "npm install -D". Do you want to install 'webpack-cli' (yes/no): npm ERR! code 1 npm ERR! path /Users/arthos/Sites/donations npm ERR! command failed npm ERR! command sh -c cross-env NODE_ENV=development node_modules/webpack/bin/webpack.js --progress --config=node_modules/laravel-mix/setup/webpack.config.js npm ERR! A complete log of this run can be found in: npm ERR! /Users/arthos/.npm/_logs/2020-11-04T08_50_44_245Z-debug.log npm ERR! code 1 npm ERR! path /Users/arthos/Sites/donations npm ERR! command failed npm ERR! command sh -c npm run development npm ERR! A complete log of this run can be found in: npm ERR! /Users/arthos/.npm/_logs/2020-11-04T08_50_44_427Z-debug.log
```

If doing `$ npm run dev` in iTerm, I get that prompt to install webpack-cli:
<img width="795" alt="Capture d’écran 2020-11-04 à 09 53 30" src="https://user-images.githubusercontent.com/2260685/98090349-d21f2800-1e7b-11eb-8ed9-289fc57c9c88.png">

And it finally fails with a Webpack configuration error: 
```
> dev
> npm run development


> development
> cross-env NODE_ENV=development node_modules/webpack/bin/webpack.js --progress --config=node_modules/laravel-mix/setup/webpack.config.js

[webpack-cli] Invalid configuration object. Webpack has been initialized using a configuration object that does not match the API schema.
 - configuration.module.rules[10] has an unknown property 'loaders'. These properties are valid:
   object { compiler?, dependency?, descriptionData?, enforce?, exclude?, generator?, include?, issuer?, loader?, mimetype?, oneOf?, options?, parser?, realResource?, resolve?, resource?, resourceFragment?, resourceQuery?, rules?, sideEffects?, test?, type?, use? }
   -> A rule description with conditions and effects for modules.
npm ERR! code 2
npm ERR! path /Users/arthos/Sites/donations
npm ERR! command failed
npm ERR! command sh -c cross-env NODE_ENV=development node_modules/webpack/bin/webpack.js --progress --config=node_modules/laravel-mix/setup/webpack.config.js

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/arthos/.npm/_logs/2020-11-04T08_55_40_364Z-debug.log
npm ERR! code 2
npm ERR! path /Users/arthos/Sites/donations
npm ERR! command failed
npm ERR! command sh -c npm run development

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/arthos/.npm/_logs/2020-11-04T08_55_40_557Z-debug.log
```

Good work, there's progress! 😄

**Comment by taliesinpenbardd** [comment]
Didn't mean to close the issue...

**Comment by TiagoSilvaPereira** [comment]
Hi @taliesinpenbardd it seems like an issue with Laravel Mix... please try these instructions from the mix repository: https://github.com/JeffreyWay/laravel-mix/issues/16

Something like it may work:
```
npm uninstall laravel-mix && npm install --save laravel-mix
```
Or you can try it:

```
rm -r node_modules
rm package.json
npm cache clean -f
npm install
```

Yesterday the Laravel Team changed the master branch to point to the upcoming Laravel 9... so I needed to change Vemto to clone Laravel 8 from a new branch. If you are testing in a project you generated yesterday... I would recommend deleting the project folder then generating again because it probably downloaded the wrong Laravel Mix version.

As the issue is probably not related to Vemto, I'm closing here. But feel free to comment or open another issue if you have another problem.

Cheers

**Comment by taliesinpenbardd** [comment]
I did that, and erased and generated again the project, but I'm stuck on the same stage with the same errors.
Just to be sure, I created a new project with `laravel new donations2`, then both `yarn && yarn dev` and  `npm install && npm run dev` run smoothly (I obviously removed the node_modules folder and the yarn.lock file before trying the npm command).
Well, I'll test it later. Have a nice day!
